WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductInOrderViewModel.swift

```diff
@@ -1,39 +1,71 @@
 import Yosemite
+import Combine
+import WooFoundation
 
 /// View model for `ProductInOrder`.
 ///
 final class ProductInOrderViewModel: Identifiable {
+    /// Encapsulates the necessary information to execute adding discounts to products
+    /// 
+    struct DiscountConfiguration {
+        let addedDiscount: Decimal
+        let baseAmountForDiscountPercentage: Decimal
+        let onSaveFormattedDiscount: (String?) -> Void
+    }
+
     /// The product being edited.
     ///
     let productRowViewModel: ProductRowViewModel
 
+    let addedDiscount: Decimal
+
+    var formattedDiscount: String? {
+        currencyFormatter.formatAmount(addedDiscount)
+    }
+
     let baseAmountForDiscountPercentage: Decimal
 
     /// Closure invoked when the product is removed.
     ///
     let onRemoveProduct: () -> Void
 
-    let isAddingDiscountToProductEnabled: Bool
+    private let isAddingDiscountToProductEnabled: Bool
+
+    var showAddDiscountRow: Bool {
+        isAddingDiscountToProductEnabled && addedDiscount == 0
+    }
+
+    var showCurrentDiscountSection: Bool {
+        isAddingDiscountToProductEnabled && addedDiscount != 0
+    }
 
     let onSaveFormattedDiscount: (String?) -> Void
 
+    var viewDismissPublisher = PassthroughSubject<(), Never>()
+
+    private let currencyFormatter: CurrencyFormatter
+
     init(productRowViewModel: ProductRowViewModel,
-         baseAmountForDiscountPercentage: Decimal,
-         onRemoveProduct: @escaping () -> Void,
-         onSaveFormattedDiscount: @escaping (String?) -> Void,
-         isAddingDiscountToProductEnabled: Bool = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.ordersWithCouponsM4)) {
+         productDiscountConfiguration: DiscountConfiguration?,
+         storeCurrencySettings: CurrencySettings = ServiceLocator.currencySettings,
+         onRemoveProduct: @escaping () -> Void) {
         self.productRowViewModel = productRowViewModel
-        self.baseAmountForDiscountPercentage = baseAmountForDiscountPercentage
+        self.addedDiscount = productDiscountConfiguration?.addedDiscount ?? .zero
+        self.baseAmountForDiscountPercentage = productDiscountConfiguration?.baseAmountForDiscountPercentage  ?? .zero
         self.onRemoveProduct = onRemoveProduct
-        self.onSaveFormattedDiscount = onSaveFormattedDiscount
-        self.isAddingDiscountToProductEnabled = isAddingDiscountToProductEnabled
+        self.onSaveFormattedDiscount = productDiscountConfiguration?.onSaveFormattedDiscount ?? { _ in }
+        self.isAddingDiscountToProductEnabled = productDiscountConfiguration != nil
+        self.currencyFormatter = CurrencyFormatter(currencySettings: storeCurrencySettings)
     }
 
     lazy var discountDetailsViewModel: FeeOrDiscountLineDetailsViewModel = {
-        FeeOrDiscountLineDetailsViewModel(isExistingLine: false,
+        FeeOrDiscountLineDetailsViewModel(isExistingLine: addedDiscount > 0,
```

Comment:
I'm not sure how much of an edge case this could be, but I recall we could expect positive discounts? I could be mistaken, but if that's the case, should we decorate this property as `abs(addedDiscount) > 0` ?

Reply:
Good catch, changed in 69171851f3558d38a417f67d4820ae81cc9aa975